src/main/java/com/labsynch/labseer/domain/StandardizationDryRunCompound.java

```diff
@@ -909,5 +909,44 @@ public static Collection<StandardizationDryRunCompound> fromJsonArrayToStandardi
 				.use("values", StandardizationDryRunCompound.class).deserialize(json);
 	}
 
+	@Transactional
+	public static List<Long> fetchUnlockedParentIds(int limit) {
+		String sql = "SELECT p.id " +
+					"FROM parent p " +
+					"WHERE NOT EXISTS (" +
+					"    SELECT 1 " +
+					"    FROM standardization_dry_run_compound s " +
+					"    WHERE s.parent_id = p.id" +
+					") " +
+					"ORDER BY p.id ASC " +
+					"FOR UPDATE SKIP LOCKED " +
+					"LIMIT :limit";
+
+		@SuppressWarnings("unchecked")
+		List<BigInteger> result = StandardizationDryRunCompound.entityManager()
+				.createNativeQuery(sql)
+				.setParameter("limit", limit)
+				.getResultList();
+
+		// Convert BigInteger to Long
+		return result.stream()
+					.map(BigInteger::longValue)
+					.collect(Collectors.toList());
+	}
+
+	@Transactional
+	public static int countRemainingParentsNotInStandardizationDryRunCompound() {
```

Comment:
This second query is important because it's executed to determine how many rows are left to be added to the table.  It's possible that e.g. the final batch of checked out parent ids fail for some reason.  If thats the case, we need a final query to verify that all the ids are in the table before existing the dry run process.